docs/dev/experimental-mode.adoc

```diff
@@ -19,74 +19,104 @@ Experimental mode can be enabled by setting odo preferences as well.
+wildfly                    openshift     10.0,10.1,11.0,12.0,13.0,8.1,9.0,latest     NO
 
-If experimental mode is not enabled, `--devfile` flag is not seen in the `odo help push`
-```sh
-$ odo push -h
-Push source code to a component.
-
-Usage:
-  odo push [component name] [flags]
-
-Examples:
-  # Push source code to the current component
-  odo push
-  
-  # Push data to the current component from the original source.
-  odo push
-  
-  # Push source code in ~/mycode to component called my-component
-  odo push my-component --context ~/mycode
-
-Flags:
-      --config               Use config flag to only apply config on to cluster
-      --context string       Use given context directory as a source for component settings
-  -f, --force-build          Use force-build flag to force building the component
-  -h, --help                 Help for push
-      --ignore stringSlice   Files or folders to be ignored via glob expressions.
-      --show-log             If enabled, logs will be shown when built
-      --source               Use source flag to only push latest source on to cluster
-
-Additional Flags:
-  -v, --v Level              Log level for V logs. Level varies from 0 to 9 (default 0).
-      --vmodule moduleSpec   Comma-separated list of pattern=N settings for file-filtered logging
+Odo Devfile Components:
+NAME                 DESCRIPTION                           SUPPORTED
+maven                Upstream Maven and OpenJDK 11         YES
+nodejs               Stack with NodeJS 10                  YES
+openLiberty          Open Liberty microservice in Java     YES
+java-spring-boot     Spring Boot® using IBM Java           YES
 ```
 
-If the experimental mode is disabled and the hidden flag is used, it will throw an unknown flag error.
-```sh
-$ odo push --devfile ./devfile.yaml  
-Error: unknown flag: --devfile
+##### Creating a devfile component:
+When creating a devfile component with the experimental flag enabled, a local config of the component settings in created at `.odo/env/env.yaml` in the project directory.
+
+```
+$ odo create java-spring-boot myspring
+Experimental mode is enabled, use at your own risk
+
+Validation
+ ✓  Checking Devfile compatibility [234445ns]
+ ✓  Validating Devfile [84112ns]
+
+Please use `odo push` command to create the component with source deployed
+```
+
+##### Creating and deleting component url:
+Create a devfile component url with `odo url create` and delete a devfile component url with `odo url delete`. This updates the component settings at `.odo/env/env.yaml` accordingly.
+
+When creating a component url, the ingress domain must be specified with the flag `--host` and a valid port must be specified with `--port`. The ports to be pushed, must be in the devfile.
+
+```
+$ odo url create myspring --port 8080 --host <ingress domain>
+ ✓  URL created for component: myspring, cluster host: <ingress domain>
+
+To create URL on the OpenShift Cluster, please use `odo push`
```

Comment:
Is this OpenShift specific?  Should we say something like `the cluster in Kubernetes` instead of `OpenShift Cluster`?

Reply:
We should make it just `cluster`, as Tomas suggested earlier.